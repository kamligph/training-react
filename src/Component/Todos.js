import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TodoItem from './../Component/TodoItem';

class Todos extends Component {
  render() {
    const todoLimit = 50;
    let todoItems;
    if(this.props.todos) {
      todoItems = this.props.todos.slice(0, todoLimit).map(
        todo => {
          return (
            <TodoItem key={todo.id} todo={todo} />
          );
        }
      );
    }

    return (
      <div className="Todos">
        <h3>To Dos</h3>
        {todoItems}
      </div>
    );
  }
}

Todos.propTypes = {
  todos: PropTypes.array
}

export default Todos;
