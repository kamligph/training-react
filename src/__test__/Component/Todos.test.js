import React from 'react';
import axios from 'axios';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import Todos from './../../Component/Todos'; 

describe('[Todo.js]', () => {
  it('Renders without crashing', () => {
    const component = renderer.create(
      <Todos />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it('Get promised todos v1', async() => {    
  //   expect.assertions(1);
  //   await axios.get('https://jsonplaceholder.typicode.com/todos')
  //     .then((response) => {
  //       expect(response.data.length).toBe(200);
  //     })
  // });

  it('Check proptypes', async() => {    
    // let todos;
    // await axios.get('https://jsonplaceholder.typicode.com/todos')
    //   .then((response) => {
    //     todos = response.data
    //   })
    
    // const wrapper = shallow(
    //   <Todos todos={todos} />
    // );

    // expect(wrapper.state('todos').length).toBe(5)

  });
});
