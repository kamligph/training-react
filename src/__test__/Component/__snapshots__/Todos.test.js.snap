// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Todo.js] Renders without crashing 1`] = `
<div
  className="Todos"
>
  <h3>
    To Dos
  </h3>
</div>
`;
